CriticalMass/BlurryOverlayView.swift

```diff
@@ -8,15 +8,74 @@
 
 import UIKit
 
+@IBDesignable
+class RoundedButton: UIButton {
+    override func awakeFromNib() {
+        configureUI()
+    }
+
+    override func prepareForInterfaceBuilder() {
+        configureUI()
+    }
+
+    private func configureUI() {
+        setTitleColor(.black, for: .normal)
+        backgroundColor = .yellow100
+        layer.cornerRadius = 24
+        titleLabel?.font = UIFont.scalableSystemFont(fontSize: 17, weight: .bold)
+        titleLabel?.adjustsFontForContentSizeCategory = true
+    }
+}
+
 class BlurryOverlayView: UIView, IBConstructable {
-    @IBOutlet private var messageLabel: UILabel!
+    @objc
+    dynamic var gradientBeginColor: UIColor = .black {
+        didSet {
+            updateGradient()
+        }
+    }
 
-    public var message: String? {
+    @objc
+    dynamic var gradientEndColor: UIColor = .white {
         didSet {
-            guard let text = message else {
-                return
-            }
-            messageLabel.text = text
+            updateGradient()
+        }
+    }
+
+    @IBOutlet private var titleLabelTopConstraint: NSLayoutConstraint!
+    @IBOutlet private var settingsButton: RoundedButton!
+    @IBOutlet private var messageLabel: UILabel!
+    @IBOutlet private var titlelabel: UILabel!
+
+    override class var layerClass: AnyClass {
+        return CAGradientLayer.self
+    }
+
+    override func awakeFromNib() {
+        updateGradient()
+    }
+
+    public func set(title: String, message: String) {
+        titlelabel.text = title
+        messageLabel.text = message
+    }
+
+    @IBAction private func didTapSettingsButton(_: Any) {
+        UIApplication.shared.open(URL(string: UIApplication.openSettingsURLString)!)
```

Comment:
Hmm. Maybe we should add a controller for the BlurryOverlay, since it now does logic. wdyt?

Reply:
Or we use the responderchain and set the action on the undelying controller?

Reply:
its no a lot of logic but you are right. I'll let the controller handle the action

Reply:
True. Very little logic